.github/workflows/go-build-and-test.yml

```
@@ -0,0 +1,25 @@
+name: build-and-test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
```

The project is based on go 1.20. (When in doubt, check go.mod)
```suggestion
        go-version: 1.20
```

---

ACK